src/js/program/shared-media-pool.ts

```diff
@@ -0,0 +1,20 @@
+import { MediaElementPoolInt, OptionalUndefVideo } from 'program/media-element-pool';
+
+export default function SharedMediaPool(sharedElement: HTMLVideoElement, mediaPool: MediaElementPoolInt): MediaElementPoolInt {
+    return Object.assign({}, mediaPool, {
+        prime(): void {
+            if (!sharedElement.src) {
+                sharedElement.load();
+            }
+        },
+        getPrimedElement(): OptionalUndefVideo {
+            return sharedElement;
```

Comment:
`sharedElement` is always and `HTMLVideoElement`.

Is the return type here just used to match `MediaElementPool .getPrimedElement` ? Why?

Reply:
yeah i was matching - was unclear on use case for the func, updated!